Tests/Unit/MSFT_xWebAppPool.Tests.ps1

```diff
@@ -0,0 +1,862 @@
+        {
+            <#
+
+            .SYNOPSIS
+            Runs Set-TargetResource on $option and then verifies that option was changed to the desired value.  Returns $true if option was changed.
+
+            .DESCRIPTION
+            Runs Set-TargetResource on $option and then verifies that option was changed to the desired value.  Returns $true if option was changed.
+
+            .PARAMETER option 
+            should be the name of the option as defined in xWebAppPool.psm1
+
+            .PARAMETER appcmdVal
+            The location that would be used to get this value via appcmd.
+            
+            .PARAMETER value1
+            A valid value that $option can be set to.  Must be different than value2
+
+            .PARAMETER value2
+            A valid value that $option can be set to.  Must be different than value1
+
+            .EXAMPLE
+            SetTargetResource "autoStart" "add.autoStart" "true" "false"
+
+            #>
+            [OutputType([bool])]
+            Param($option,$appcmdVal,$value1,$value2)
+            $testParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+            $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+            if($curVal -ne $value1)
+            {
+                $testParams.Add($option,$value1)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value1 -eq $curVal
+            }
+            else
+            {
+                $testParams.Add($option,$value2)
+                Set-TargetResource @testParams | Out-Null #So we only return the value of Test-TargetResource below
+                $curVal = Invoke-Expression "(([xml](& $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool 'DefaultAppPool' /config:*)).$($appcmdVal))"
+                $value2 -eq $curVal
+            }
+
+            Return 
+        }
+
+        try
+        {
+            $baseParams =@{
+                Name = "DefaultAppPool"
+                Ensure = "Present"
+            }
+
+            It 'DefaultAppPool should exist' {           
+                (Get-ChildItem IIS:\apppools).Name.Contains("DefaultAppPool") | Should be $true
+            }
+
+            Context 'Test-TargetResource' {
+                
+                It 'Passes autoStart Test when same' {
+                    
+                    TestTargetResourceSame "autoStart" "add.autoStart" | Should be $true
+                }
+
+                It 'Fails autoStart Test when different' {
+                    
+                    TestTargetResourceDiff "autoStart" "add.autoStart" "true" "false"| Should be $false
```

Comment:
This can become:
Test-TargetResource -AutoStart $false | Should Be $false

Again because $global:PoolCfg has autostart = $true this should return false without us having to do much more.